lib/react.js

```diff
@@ -15,17 +15,62 @@ module.exports = {
     },
   },
   rules: {
-    'react/jsx-indent': 'off',
-    'react/jsx-space-before-closing': 'off',
+    // Enforce JSX indentation
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-indent.md
+    'react/jsx-indent': ['error', 2],
+
+    // Enforce spacing before closing bracket in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-space-before-closing.md
+    'react/jsx-space-before-closing': ['off', 'always'],
+
+    // Disallow React to be incorrectly marked as unused
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-uses-react.md#disallow-react-to-be-incorrectly-marked-as-unused-reactjsx-uses-react
     'react/jsx-uses-react': 'error',
+
+    // Disallow variables used in JSX to be incorrectly marked as unused
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md
     'react/jsx-uses-vars': 'error',
-    'react/require-default-props': ['warn', { forbidDefaultForRequired: true }],
-    'react/forbid-prop-types': [
-      'warn',
-      { forbid: ['any', 'array', 'object'], checkContextTypes: false, checkChildContextTypes: false },
-    ],
+
+    // Enforce a defaultProps definition for every prop that is not a required prop
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/843d71a432baf0f01f598d7cf1eea75ad6896e4b/docs/rules/require-default-props.md
+    'react/require-default-props': ['error', {
+      forbidDefaultForRequired: true,
+    }],
+
+    // Forbid certain propTypes (any, array, object)
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/843d71a432baf0f01f598d7cf1eea75ad6896e4b/docs/rules/forbid-prop-types.md
+    'react/forbid-prop-types': ['error', {
+      forbid: ['any', 'array', 'object'],
+      checkContextTypes: true,
+      checkChildContextTypes: true,
+    }],
+
     'react-hooks/rules-of-hooks': 'error',
     'react-hooks/exhaustive-deps': 'warn',
     'react/jsx-handler-names': 'warn',
+
+    // Enforce a specific function type for function components
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md
+    'react/function-component-definition': ['error', {
+      namedComponents: ['function-declaration', 'function-expression'],
+      unnamedComponents: 'function-expression',
+    }],
+
+    // Prevent unused propType definitions
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md
+    'react/no-unused-prop-types': ['error', {
+      customValidators: [],
+      skipShapeProps: true,
+    }],
```

Comment:
I think we don't need `customValidators` since it's optional and `skipShapeProps` is `true` by default.
```suggestion
    'react/no-unused-prop-types': 'error',
```